src/api/comment/content-types/comment/lifecycles.js

```diff
@@ -4,7 +4,7 @@ module.exports = {
     // Set author to the user sending the request
     const ctx = strapi.requestContext.get();
     event.params.data.authorId = ctx.state.user;
```

Comment:
If `ctx.state.user` is the authorId, I would think the type is an integer. But `event.params.data.author = ctx.state.user.username` suggests its an object. Maybe this should be `ctx.state.user.Id` or some other field?

Reply:
Yeah, I originally thought similarly, to use just user.id, but for reference strapi needs the whole object. Since `author` already taken and we need to keep it as a string for backward compatibility maybe I should have a different field name to avoid confusion. `commenter` or `user` ? 🤔

Reply:
Ended up using `user`